react/tic-tac-toe/src/redux/match/reducer.js

```diff
@@ -0,0 +1,38 @@
+import { getWinner } from '../../app/screens/Game/utils';
+
+import { actions } from './actions';
+
+const initialState = {
+  history: [{
+    squares: Array(9).fill(null)
+  }],
+  stepNumber: 0,
+  xIsNext: true,
+  matches: [],
+  isLoading: false
+};
+
+function reducer(state = initialState, action) {
+  switch (action.type) {
+    case actions.SET_BOARD: {
+      const { history, stepNumber, xIsNext } = state;
+      const newHistory = history.slice(0, stepNumber + 1);
```

Comment:
I think there is no need to make a newHistory array 👀You are copying the current history, but not adding or removing anything else. You can do the same steps with just the current history.

```js
const squares = [...history[history.length - 1].squares];
if (getWinner(squares) || squares[action.payload]) return state; 
squares[action.payload] = xIsNext ? 'X' : 'O';
return { ...state, history: [...history, { squares }], stepNumber: stepNumber + 1, xIsNext: !xIsNext };
```

Reply:
mm but i dont need the last element from the history array, i need the element at the 'stepNumber' position.